ElectricityBillApi/Models/ElectricProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace ElectricityBillApi.Models
+{
+    public class ElectricProvider
+    {
+        private PowerPlant _subscibedPowerPlant;
+
+        public string Name { get; set; }
+
+        public decimal CalculatePrice(Address address)
+        {
+            if (_subscibedPowerPlant == default) throw new NotImplementedException("What to do here? More infos needed!");
```

Comment:
Oh, but it is implemented. Instead, thow `ArgumentNullException` and provide the null argument name using `nameof`.

Reply:
Hmm, in my mind I thought I don't want exceptions in a working program, so I should cover all possible flaws with good buisness logic...